index.js

```diff
@@ -75,9 +79,17 @@ function snapshot(module) {
  * Add a module entry back into the cache
  * @param module module cache object
  */
+var restoredModules = [];
 function restore(module) {
     require.cache[module.id] = module;
-    module.children.forEach(restore);
+    var temp = restoredModules.slice();
+    module.children.forEach(function (child) {
+      if(!restoredModules.includes(child.id)){
```

Comment:
Array.prototype.includes does not exists on node 0.12 or node 0.10 (don't know about io.js),when travis runs the tests on this node versions it will be undefined and the tests will fail.

You may want to deprecate node 4 and previous versions or change the function for something that exists.